openfecli/parameters/mol.py

```diff
@@ -0,0 +1,36 @@
+# This code is part of OpenFE and is licensed under the MIT license.
+# For details, see https://github.com/OpenFreeEnergy/openfe
+
+from plugcli.params import MultiStrategyGetter, Option, NOT_PARSED
+
+
+def _load_molecule_from_smiles(user_input, context):
+    from openfe.setup import Molecule
+    from rdkit import Chem
+    # MolFromSmiles returns None if the string is not a molecule
+    # TODO: find some way to redirect the error messages? Messages stayed
+    # after either redirect_stdout or redirect_stderr.
+    mol = Chem.MolFromSmiles(user_input)
+    if mol is None:
+        return NOT_PARSED
+
+    # TODO: next is (temporary?) hack: see
+    # https://github.com/OpenFreeEnergy/Lomap/issues/4
+    Chem.rdDepictor.Compute2DCoords(mol)
+    return Molecule(rdkit=mol)
+
+
+get_molecule = MultiStrategyGetter(
+    strategies=[
+        # NOTE: I think loading from smiles must be last choice, because
+        # failure will give meaningless user-facing errors
+        _load_molecule_from_smiles,
+    ],
+    error_message="Unable to generate a molecule from '{user_input}'."
+)
```

Comment:
So a `.` in a SMILES would mean a disconnected component, which I think we can assume we won't be given... so another strategy could be that any string with a dot is a filepath.  Is the StrategyGetter just iterating through the list of functions until it finds a hit?

Reply:
> Is the StrategyGetter just iterating through the list of functions until it finds a hit?

Basically, yeah. It attempts to parse user input according the function in each of the `strategies`, and those functions return the sentinel `plugcli.params.NOT_PARSED` if they can't interpret the input. So it's sort of a "forgiveness, not permission" approach -- if we added a `_load_molecule_from_file` strategy before `_load_molecule_from_smiles`, then it will take the user input, try to interpret it as a filename. If that fails (file doesn't exist or can't be understood) then it will fall back to `_load_molecule_from_smiles`, which would treat the same string as SMILES.

Adding new strategies is easy. I didn't include file loading in this PR just to try to keep it minimal, but my assumption is that loading from file will usually be the first choice.

Also, `get_molecule` can be replaced with a function, if we want more control on, e.g., error handling. `MultiStrategyGetter` is just a shortcut for making a callable from composable pieces.